ads/google/a4a/test/test-utils.js

```diff
@@ -630,6 +631,66 @@ describes.sandboxed('Google A4A utils', {}, (env) => {
         });
       });
     });
+
+    it('should include user agent hint params', () => {
+      return createIframePromise().then((fixture) => {
+        setupForAdTesting(fixture);
+        const {doc} = fixture;
+        doc.win = fixture.win;
+        const elem = createElementWithAttributes(doc, 'amp-a4a', {});
+        const impl = new MockA4AImpl(elem);
+        noopMethods(impl, fixture.ampdoc, env.sandbox);
+        Object.defineProperty(impl.win.navigator, 'userAgentData', {
+          'value': {
+            'getHighEntropyValues': () =>
+              Promise.resolve({
+                platform: 'Windows',
+                platformVersion: 10,
+                architecture: 'x86',
+                model: 'Pixel',
+                uaFullVersion: 3.14159,
+              }),
+          },
+        });
+        return fixture.addElement(elem).then(() => {
+          return googleAdUrl(impl, '', Date.now(), [], []).then((url) => {
+            expect(url).to.match(
+              /[&?]uap=Windows&uapv=10&uaa=x86&uam=Pixel&uafv=3.14159[&$]/
+            );
+          });
+        });
+      });
+    });
+
+    it('should proceed if user agent hint params time outs', () => {
+      return createIframePromise().then((fixture) => {
+        setupForAdTesting(fixture);
+        const clock = fakeTimers.withGlobal(fixture.win).install({
```

Comment:
did you try getting this from env object? I think `env.sandbox.useFakeTimers()` might do the same thing? https://sinonjs.org/releases/v1.17.6/fake-timers/

Reply:
FYI you dont need to do the cleanup after the test like in that linked example. It should be already handled by the test runner.

Reply:
I tried but the sandbox's timer doesn't apply to the window object in the test fixture. The test's still pausing for 1 second with `env.sandbox.useFakeTimers()`.

Reply:
ok thanks for confirming.